### DISCUSSION TITLE
Vercel not handling relative imports?

### DISCUSSION THREAD

**Discussion by YitzchakMeltz** [vercel/community#5642] [Help] [answered]
### Summary

I have an express js app deployed on Vercel. Locally, the app runs fine, however when I deploy it and try to run it I get the following error:
```
Cannot find module '../database/connection'
Require stack:
- /var/task/src/app/services/galleryService.js
- /var/task/src/app/controllers/galleryController.js
- /var/task/src/app/routes/galleryRoutes.js
- /var/task/src/app/routes/routes.js
- /var/task/src/routes/router.js
- /var/task/src/app.js
- /var/task/___now_launcher.js
- /var/runtime/index.mjs
Did you forget to add it to "dependencies" in `package.json`?
INIT_REPORT Init Duration: 278.23 ms	Phase: init	Status: error	Error Type: Runtime.ExitError
Cannot find module '../database/connection'
Require stack:
- /var/task/src/app/services/galleryService.js
- /var/task/src/app/controllers/galleryController.js
- /var/task/src/app/routes/galleryRoutes.js
- /var/task/src/app/routes/routes.js
- /var/task/src/routes/router.js
- /var/task/src/app.js
- /var/task/___now_launcher.js
- /var/runtime/index.mjs
Did you forget to add it to "dependencies" in `package.json`?
INIT_REPORT Init Duration: 313.76 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
Error: Runtime exited with error: exit status 1
Runtime.ExitError
```
The file that it says is not found is a file that I created which is located in the same repo. As per advice online, I renamed the folder and the filename as well as adjusted the imports to rule out case sensitivity issues with no avail.

Below is my file structure and the problematic import:
![image](https://github.com/vercel/community/assets/75029003/ba3dcac1-3a01-459c-ada5-cdc37b3c3f70)

### Steps to Reproduce

Not sure if this is a general problem...

**Comment by amyegan** [comment, marked as answer]
Hi @YitzchakMeltz. It looks like you're running into some problems because of the way Express.js needs to be configured to run in serverless environments. I recommend the following guides to help you get it working:
- [Using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel)
- [How can I use files in Serverless Functions on Vercel?](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions)
